src/flow.c

```diff
@@ -485,7 +485,13 @@ void FlowHandlePacketUpdate(Flow *f, Packet *p, ThreadVars *tv, DecodeThreadVars
         SCLogDebug("pkt %p FLOW_PKT_ESTABLISHED", p);
         p->flowflags |= FLOW_PKT_ESTABLISHED;
 
-        FlowUpdateState(f, FLOW_STATE_ESTABLISHED);
+        if (
+#ifdef CAPTURE_OFFLOAD
+                (f->flow_state != FLOW_STATE_CAPTURE_BYPASSED) &&
+#endif
+                (f->flow_state != FLOW_STATE_LOCAL_BYPASSED)) {
+            FlowUpdateState(f, FLOW_STATE_ESTABLISHED);
```

Comment:
should we flag the packet as part of an established flow in this case, as we do on line 486?

Reply:
I guess so.

We have
1. UDP from client to server, nothing special
2. UDP from server to client, we trigger a `bypass` rule that `FlowUpdateState(FLOW_STATE_LOCAL_BYPASSED)`, and in later processing, we call `FlowHandlePacketUpdate` that also sees the flow as established...